modules/saml/lib/IdP/SAML2.php

```diff
@@ -1300,7 +1300,13 @@ private static function encryptAssertion(
 
         $sharedKey = $spMetadata->getString('sharedkey', null);
         if ($sharedKey !== null) {
-            $key = new XMLSecurityKey(XMLSecurityKey::AES128_CBC);
+            $algo = $spMetadata->getString('sharedkey_algorithm', null);
+            if ($algo === null) {
+                // If no algorithm is configured, use a sane default
+                $algo = $idpMetadata->getString('sharedkey_algorithm', XMLSecurityKey::AES128_GCM);
```

Comment:
Uhm, this breaks backwards compatibility, right?

Reply:
I guess I did this because the old default is weak/broken... It's our legacy of not producing a new major in over a decade..
We have done these kind of BC breaks before and as long as we add proper upgrade notes I don't see an issue..
This PR is targeted at `master`, so it shouldn't be a problem.. The backport to 1.19 can be different is you like

Reply:
@jaimeperez Please get back to me on this so I can merge this

Reply:
@jaimeperez Pretty please?

Reply:
For 1.19 please default to what it was before. This is the algo originally supported by world+dog

Reply:
Sorry so much @tvdijen! For some reason my browser logged me off github and I never got notifications about this 😞 

I agree with Thijs, we should keep what we had before. The main reason for me is that GCM support in some SAML implementations is not that common yet, so defaulting to it could potentially break interoperability with other products. For 2.0 it's totally fine to change it.